crypto.go

```diff
@@ -136,11 +143,28 @@ func EncryptData(Data map[string]string, ak SymmetricKey) *map[string]string {
 	return &encryptedData
 }
 
+// DecryptSignedData decrypts and verifies signed payloads for data that has been encrypted and signed using TFSP1;ED25519;BLAKE2B
+func DecryptSignedData(data map[string]string, ak SymmetricKey, psk []byte, salt string) (*map[string]string, error) {
+	// the verify function uses the provided public signing key and salt to verify "key-message" pairs that are passed
+	// as parameters, message should be a TFSP1;ED25519;BLAKE2B formatted string. The returned value is the unsigned message only.
+	verify := func(key, message string) (string, error) {
+		verifiedMessage, err := VerifyField(key, message, psk, salt)
+		return verifiedMessage, err
+	}
+	return decryptDataWithProcessFunction(data, ak, verify)
+}
+
 // DecryptData decrypts a collection of data of string key and values encrypted using Tozny v1 Record encryption,
 // returning the decrypted data and error (if any).
-func DecryptData(Data map[string]string, ak SymmetricKey) (*map[string]string, error) {
+func DecryptData(data map[string]string, ak SymmetricKey) (*map[string]string, error) {
+	return decryptDataWithProcessFunction(data, ak, func(key string, message string) (string, error) { return message, nil })
+}
+
+// decryptDataWithProcessFunction decrypts a map of key value pairs where values are encrypted using Tozny v1 Record encryption
+// after description a provided function is run against each key-value pair. The result of the function if err = nil is set as the value
+func decryptDataWithProcessFunction(data map[string]string, ak SymmetricKey, postprocess func(key string, value string) (string, error)) (*map[string]string, error) {
```

Comment:
Why is this not a public function?

Reply:
I usually hide helper methods that I can't think of good third party use cases. But I can make it public